docs/guide/scala_guide.md

```diff
@@ -162,3 +162,61 @@ println(fory.deserialize(fory.serialize(opt)))
 val opt1: Option[Long] = None
 println(fory.deserialize(fory.serialize(opt1)))
 ```
+
+## Scala Case Class Default Values Support
+
+Fury now supports Scala case class default values during deserialization when using meta-shared serialization mode. This feature allows for forward/backward compatibility when case classes have default parameters.
+
+### Overview
+
+When a Scala case class has default parameters, the Scala compiler generates companion object methods like `apply$default$1`, `apply$default$2`, etc. that return the default values. Fury can now detect these methods and use them when deserializing objects where certain fields are missing from the serialized data.
+
+### Example
+
+Consider a Scala case class with default values:
+
+```scala
+case class SomeClass(v: List[IdAnyVal], x: Int = 1)
```

Comment:
Can we default List, Map, Set and Option to their empty values instead of null? If an explicit default is provided, use that but without one, empty values are good for these types.
Assuming 0 or "" for numbers and strings, I don't think that would be a good idea.

Reply:
We could do this, but if user's code has some special logic conditioned on the nullability, this will break their code.